app/components/menu-action-item.tsx

```diff
@@ -49,31 +49,37 @@ export type MenuActionProps = {
 export const MenuActionItem = (
   props: MenuActionProps & { as: "menuitem" | "button" }
 ) => {
-  const { label, iconName } = props;
+  const { label, trailingIconName, leadingIconName } = props;
   const {
     isOpen: isConfirmationOpen,
     open: openConfirmation,
     close: closeConfirmation,
   } = useDisclosure();
+
   const Wrapper = ({
-    icon,
+    leadingIcon,
+    trailingIcon,
     label,
     color = "primary",
+    onClick,
     ...rest
   }: {
-    icon?: IconName;
+    leadingIcon?: IconName;
+    trailingIcon?: IconName;
     label: string | NonNullable<React.ReactNode>;
     color?: MenuActionProps["color"];
+    onClick?: (e: React.MouseEvent<HTMLElement>) => void;
   }) => {
-    const handleClick = () => {
-      if (props.onClick) {
-        if ("requireConfirmation" in props && props.requireConfirmation) {
+    const handleClick = (e: React.MouseEvent<HTMLElement>) => {
+      if (props.type === "button") {
+        if (props.requireConfirmation) {
           openConfirmation();
         } else {
-          return props.onClick();
+          props.onClick(e);
         }
       }
     };
+
```

Comment:
@bprusinowski Should Link be getting `onClick`?

Reply:
I think technically it can, I am just not sure if this wouldn't break something, as also we can't preview the deployment before merging – if it's not necessary for other reason, I'd keep it, unless you are sure that every place that uses it works as before 😄